terminus-ui/chip/src/chip.component.ts

```diff
@@ -0,0 +1,432 @@
+    super(elementRef);
+
+  }
+
+  public ngOnDestroy() {
+    this.destroyed.emit({chip: this});
+    // eslint-disable-next-line no-underscore-dangle
+  }
+
+  /**
+   * Selects the chip.
+   */
+  public select(): void {
+    if (!this.selected) {
+      this.selected = true;
+      this.dispatchSelectionChange();
+    }
+  }
+
+  /**
+   * Deselects the chip.
+   */
+  public deselect(): void {
+    if (this.selected) {
+      this.selected = false;
+      this.dispatchSelectionChange();
+    }
+  }
+
+  /**
+   * Toggles the current selected state of this chip.
+   */
+  public toggleSelected(): boolean {
+    this._selected = !this.selected;
+    this.dispatchSelectionChange();
+    return this.selected;
+  }
+
+  /**
+   * Allows for programmatic focusing of the chip.
+   */
+  public focus(): void {
+    if (!this.hasFocus) {
+      this.elementRef.nativeElement.focus();
+      this.onFocus.next({chip: this});
+    }
+    this.hasFocus = true;
+  }
+
+  /**
+   * Allows for programmatic removal of the chip. Called by the TsChipCollectionComponent when the DELETE or
+   * BACKSPACE keys are pressed.
+   * {@link TsChipCollectionComponent}
+   *
+   * Informs any listeners of the removal request. Does not remove the chip from the DOM.
+   */
+  public remove(): void {
+    if (this.isRemovable) {
+      this.removed.emit({chip: this});
+    }
+  }
+
+  /**
+   * Handles click events on the chip.
+   * @param event - click event
+   */
+  public handleClick(event: Event) {
+    if (this.isDisabled) {
+      event.preventDefault();
+    } else {
+      event.stopPropagation();
+    }
+  }
+
+  /**
+   * Handle custom key presses.
+   * @param event - keyboard event
+   */
+  public handleKeydown(event: KeyboardEvent): void {
+    const code = event.code;
+    if (this.isDisabled) {
+      return;
+    }
+
+    switch (code) {
+      case KEYS.DELETE.code:
+      case KEYS.BACKSPACE.code:
+        // If it is removable, remove the focused chip
+        this.remove();
+        // Always prevent so page navigation does not occur
+        event.preventDefault();
+        break;
+      case KEYS.SPACE.code:
+        // If it is selectable, toggle the focused chip
+        if (this.isSelectable) {
+          this.toggleSelected();
+        }
+        // Always prevent space from scrolling the page since the list has focus
+        event.preventDefault();
+        break;
+      default:
+        event.preventDefault();
+        break;
+    }
+  }
+
+  public blur(): void {
+    this.ngZone.onStable
+      .asObservable()
+      .pipe(take(1))
+      .subscribe(() => {
+        this.ngZone.run(() => {
+          this.hasFocus = false;
+        });
+      });
+  }
+
+  private dispatchSelectionChange() {
+    this.selectionChange.emit(new TsChipSelectionChange(this, this._selected));
+  }
+}
+
+/**
+ * Applies proper (click) support and adds styling for use with "cancel" icon
+ *
+ * Example:
+ *
+ *     `<ts-chip>
```

Comment:
The first and last lines don't need to be indented, but all lines between need 14 space indention for our docs tool to indent properly:

![Screen Shot 2019-09-06 at 11 46 34 AM](https://user-images.githubusercontent.com/270193/64441512-06426180-d09c-11e9-85f7-dca89d821bd5.png)